common/src/main/java/org/cloudfoundry/autosleep/access/cloudfoundry/CloudFoundryApi.java

```diff
@@ -307,40 +330,45 @@ public String getHost(String routeId) throws CloudFoundryException {
                     .get(GetRouteRequest.builder()
                             .routeId(routeId)
                             .build())
-                    .get(Config.CF_API_TIMEOUT);
+                    .block(Config.CF_API_TIMEOUT);
             RouteEntity routeEntity = response.getEntity();
             String route = routeEntity.getHost() + routeEntity.getPath();
             log.debug("route =  {}", route);
-
-            GetDomainResponse domainResponse = cfClient.domains()
-                    .get(GetDomainRequest.builder()
-                            .domainId(routeEntity.getDomainId())
-                            .build())
-                    .get(Config.CF_API_TIMEOUT);
-            log.debug("domain = {}", domainResponse.getEntity());
-            return route + "." + domainResponse.getEntity().getName();
+            String domainId = routeEntity.getDomainId();
+            String domain = getDeprecatedDomain(domainId);
+            return route + "." + domain;
         } catch (RuntimeException r) {
             throw new CloudFoundryException(r);
         }
     }
 
+    private String getDeprecatedDomain(String domainId) {
+        GetDomainResponse domainResponse = cfClient.domains()
```

Comment:
Yes, I started coding calls to public and private domain API into https://github.com/cloudfoundry-community/autosleep/pull/279/commits/7064d8c99682f7c6eb8a231fa554d2c790e119f3#diff-de6409188099e3aca52a4e97bd7af702R359 but this still needs refinement:
- complete unit tests
- consider parallel non blocking calls

so I suggest we handle this into a distinct PR